### PR TITLE
Clarify when 'failed' is not a terminal RTCIceTransportState.

### LINKED ISSUE #2273: ICE restart and terminal RTCIceTransport.state "failed" state
If ICE restart results in "failed" not being terminal, we should clarify this.
If it is in fact terminal, but the transport gets swapped out, we should clarify this.

### PR DESCRIPTION
**Pull request by henbos**
Fixes #2273

***
<a href="https://pr-preview.s3.amazonaws.com/henbos/webrtc-pc/pull/2321.html" title="Last updated on Oct 4, 2019, 12:19 PM UTC (49ef27d)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/webrtc-pc/2321/e7a465d...henbos:49ef27d.html" title="Last updated on Oct 4, 2019, 12:19 PM UTC (49ef27d)">Diff</a>

### CODE REVIEW

File: webrtc.html [hunk 1]
```diff
@@ -9158,11 +9158,16 @@ <h3><dfn>RTCIceTransportState</dfn> Enum</h3>
...
+              This is a terminal state unless ICE is restarted. Since an ICE
+              restart may cause connectivity to resume, entering the
+              <code>failed</code> state does not cause DTLS transports, SCTP
+              associations or the data channels that run over them to close, or
+              tracks to mute.</td>
+              <div class="note">
+                An ICE restart can happen due to the <code>restartIce()</code>
+                method or the remote endpoint triggering renegotiation with new
+                ICE credentials.
+              </div>
```
Comment (dontcallmedom) [inline webrtc.html]: this needs to be moved inside the `<td></td>`

Comment (henbos) [reply webrtc.html]: Ack!

File: webrtc.html [hunk 2]
```diff
@@ -9158,11 +9158,16 @@ <h3><dfn>RTCIceTransportState</dfn> Enum</h3>
               gathering, received an indication that there are no more remote
               candidates, finished checking all candidate pairs, and all pairs
               have either failed connectivity checks or have lost consent.
-              This is a terminal state.  Since an ICE restart may cause
-              connectivity to resume, entering the <code>failed</code> state
-              does not cause DTLS transports, SCTP associations or the data
-              channels that run over them to close, or tracks to mute.
-              </td>
+              This is a terminal state unless ICE is restarted. Since an ICE
```
Comment (henbos) [inline webrtc.html]: unless -> until

Comment (jan-ivar) [inline webrtc.html]: s/unless/until/ ?

File: webrtc.html (hunk 1 above)
Comment (henbos) [reply webrtc.html]: Will remove note per jan-ivar's comment

**Pull request merged**